File: artifacts/tulipindicators/translate/src/ln.rs
use crate::*;
use std::f64;

pub fn ti_ln_start(options: Option<&f64>) -> i32 {
    let _ = options; // Explicitly ignore the parameter
    0
}

pub fn ti_ln(size: usize, inputs: &[&[f64]], options: &[f64], outputs: &mut [&mut [f64]]) -> i32 {
    let in1 = inputs[0];
    let output = &mut outputs[0];
    
    for i in 0..size {
        output[i] = f64::ln(in1[i]);
    }
    
    0
}

File: artifacts/tulipindicators/decompose/src/ln.c
int ti_ln_start(const double *options)
{
  (void) options;
  return 0;
}


/*** DEPENDENCIES:
***/


int ti_ln(int size, const double * const *inputs, const double *options, double * const *outputs)
{
  const double *in1 = inputs[0];
  (void) options;
  double *output = outputs[0];
  int i;
  for (i = 0; i < size; i += 1)
  {
    output[i] = log(in1[i]);
  }

  return 0;
}


/*** DEPENDENCIES:
***/


